core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -4470,7 +4470,9 @@ private void processJoinContext(
             // Key types have too much exactly except SYMBOL and STRING special case
             int columnTypeA = slaveMetadata.getColumnType(listColumnFilterA.getColumnIndexFactored(k));
             int columnTypeB = masterMetadata.getColumnType(listColumnFilterB.getColumnIndexFactored(k));
-            if (columnTypeB != columnTypeA && !(ColumnType.isSymbolOrString(columnTypeB) && ColumnType.isSymbolOrString(columnTypeA))) {
```

Comment:
Comment is misleading as there's no change for double column type . 
Please use ColumnType constants instead of raw values .